pytest-embedded-idf/pytest_embedded_idf/idf_unity_tester.py

```diff
@@ -0,0 +1,448 @@
+import time
+from threading import Semaphore, Thread
+from typing import List, Union
+
+from pexpect.exceptions import TIMEOUT
+from pytest_embedded import Dut, unity, utils
+from pytest_embedded_idf.dut import UnittestMenuCase
+
+DEFAULT_RETRY_COUNT = 3
+DEFAULT_TIMEOUT = 90
+
+
+class BaseTester:
+    """
+    The base class that providing shared methods
+
+    Attributes:
+        dut (Dut): Object of the Device under test
+        test_menu (List[UnittestMenuCase]): The list of the cases
+    """
+
+    # The patterns that indicate the runner is ready come from 'unity_runner.c'
+    ready_pattern_list = [
+        'Press ENTER to see the list of tests',
+        'Enter test for running',
+        'Enter next test, or \'enter\' to see menu',
+    ]
+
+    def __init__(self, dut: Union[Dut, List[Dut]], **kwargs) -> None:  # type: ignore
```

Comment:
```suggestion
    def __init__(self, dut: IdfDut, **kwargs) -> None:  # type: ignore
```

Reply:
I guess here it still has to be Union: 
`def __init__(self, dut: Union[IdfDut, List[IdfDut]], **kwargs) -> None:  # type: ignore`

Reply:
well, I think only `MultiDevTester` requires multiple duts?

let's keep it as what it is right now, and improve later.